network/shuffle/src/main/java/org/apache/spark/network/shuffle/ExternalShuffleClient.java

```diff
@@ -138,6 +139,26 @@ public void registerWithShuffleServer(
     client.sendRpcSync(registerMessage, 5000 /* timeoutMs */);
   }
 
+  /**
+   * Removes this application from the external shuffle server and optionally deletes local
+   * files.
+   *
+   * @param host Host of the shuffle server.
+   * @param port Port of the shuffle server.
+   * @param cleanupLocalDirs True if corresponding shuffle files should be deleted
+   * @throws IOException
+   */
+  public void applicationRemoved(String host,
+      int port,
```

Comment:
Note that other (all?) methods in this class take a host and a port. This is not a callback. This is a client wrapper for connecting to the ExternalShuffleClient. They can be on any executor. How would you tell the client where to connect to?